xcrypto/hash_test.go

```diff
@@ -208,10 +208,7 @@ func TestHash_Clone(t *testing.T) {
 			}
 			// We don't define an interface for the Clone method to avoid other
 			// packages from depending on it. Use type assertion to call it.
-			h2, err := h.(interface{ Clone() (hash.Hash, error) }).Clone()
-			if err != nil {
-				t.Fatal(err)
```

Comment:
This test should have failed before this PR, as the hash objects didn't implement the Clone method. I guess it is being incorrectly skipped due to the previous `if _, ok := h.(encoding.BinaryMarshaler); !ok {` check. It can be removed.